### ISSUE TITLE
Zend Session Exception

### ISSUE THREAD

**Issue by jasonrhodes** [frapi/frapi#85]
After setup, I can't get the admin or frontend to load at all. Frontend just returns a 500 server error, but admin gives me a bunch of PHP Notices followed by one Fatal Error, seen below:

```
Fatal error: Uncaught exception 'Zend_Session_Exception' with message 'Session must be started before any output has been sent to the browser; output started in /Users/jason/frapi/src/frapi/library/Zend/Loader/PluginLoader.php/384' in /Users/jason/frapi/src/frapi/library/Zend/Session.php:454
Stack trace:
#0 /Users/jason/frapi/src/frapi/library/Zend/Session/Namespace.php(143): Zend_Session::start(true)
#1 /Users/jason/frapi/src/frapi/admin/application/Bootstrap.php(30): Zend_Session_Namespace->__construct('locale')
#2 /Users/jason/frapi/src/frapi/library/Zend/Application/Bootstrap/BootstrapAbstract.php(662): Bootstrap->_initLanguages()
#3 /Users/jason/frapi/src/frapi/library/Zend/Application/Bootstrap/BootstrapAbstract.php(615): Zend_Application_Bootstrap_BootstrapAbstract->_executeResource('languages')
#4 /Users/jason/frapi/src/frapi/library/Zend/Application/Bootstrap/BootstrapAbstract.php(579): Zend_Application_Bootstrap_BootstrapAbstract->_bootstrap(NULL)
#5 /Users/jason/frapi/src/frapi/library/Zend/Application.php(355) in /Users/jason/frapi/src/frapi/library/Zend/Session.php on line 454
```

Thoughts on this? (Also submitted this to the Google frapi-dev group.)

Added the notices here:

```
Notice: require_once(): 1. h->opened_path=[/Users/jason/frapi/src/frapi/admin/application/Bootstrap.php]  h->filename=[/Users/jason/frapi/src/frapi/admin/application/Bootstrap.php]
 in /Users/jason/frapi/src/frapi/library/Zend/Application.php on line 320

Notice: require_once(): apc_cache_find [257072932]
 in /Users/jason/frapi/src/frapi/library/Zend/Application.php on line 320

Notice: include_once(): 1. h->opened_path=[/Users/jason/frapi/src/frapi/library/Zend/Application/Bootstrap/Bootstrap.php]  h->filename=[/Users/jason/frapi/src/frapi/library/Zend/Application/Bootstrap/Bootstrap.php]
 in /Users/jason/frapi/src/frapi/library/Zend/Loader.php on line 146

Notice: include_once(): apc_cache_find [257073436]
 in /Users/jason/frapi/src/frapi/library/Zend/Loader.php on line 146

Notice: include_once(): 1. h->opened_path=[/Users/jason/frapi/src/frapi/library/Zend/Application/Bootstrap/BootstrapAbstract.php]  h->filename=[/Users/jason/frapi/src/frapi/library/Zend/Application/Bootstrap/BootstrapAbstract.php]
 in /Users/jason/frapi/src/frapi/library/Zend/Loader.php on line 146

Notice: include_once(): apc_cache_find [257073437]
 in /Users/jason/frapi/src/frapi/library/Zend/Loader.php on line 146

Notice: include_once(): 1. h->opened_path=[/Users/jason/frapi/src/frapi/library/Zend/Application/Bootstrap/Bootstrapper.php]  h->filename=[/Users/jason/frapi/src/frapi/library/Zend/Application/Bootstrap/Bootstrapper.php]
 in /Users/jason/frapi/src/frapi/library/Zend/Loader.php on line 146

Notice: include_once(): apc_cache_find [257073438]
 in /Users/jason/frapi/src/frapi/library/Zend/Loader.php on line 146

Notice: include_once(): 1. h->opened_path=[/Users/jason/frapi/src/frapi/library/Zend/Application/Bootstrap/ResourceBootstrapper.php]  h->filename=[/Users/jason/frapi/src/frapi/library/Zend/Application/Bootstrap/ResourceBootstrapper.php]
 in /Users/jason/frapi/src/frapi/library/Zend/Loader.php on line 146

Notice: include_once(): apc_cache_find [257073440]
 in /Users/jason/frapi/src/frapi/library/Zend/Loader.php on line 146

Notice: include_once(): 1. h->opened_path=[/Users/jason/frapi/src/frapi/library/Zend/Loader/PluginLoader.php]  h->filename=[/Users/jason/frapi/src/frapi/library/Zend/Loader/PluginLoader.php]
 in /Users/jason/frapi/src/frapi/library/Zend/Loader.php on line 146

Notice: include_once(): apc_cache_find [257074608]
 in /Users/jason/frapi/src/frapi/library/Zend/Loader.php on line 146

Notice: include_once(): 1. h->opened_path=[/Users/jason/frapi/src/frapi/library/Zend/Loader/PluginLoader/Interface.php]  h->filename=[/Users/jason/frapi/src/frapi/library/Zend/Loader/PluginLoader/Interface.php]
 in /Users/jason/frapi/src/frapi/library/Zend/Loader.php on line 146

Notice: include_once(): apc_cache_find [257074611]
 in /Users/jason/frapi/src/frapi/library/Zend/Loader.php on line 146

Notice: include_once(): 1. h->opened_path=[/Users/jason/frapi/src/frapi/library/Zend/Application/Resource/Modules.php]  h->filename=[/Users/jason/frapi/src/frapi/library/Zend/Application/Resource/Modules.php]
 in /Users/jason/frapi/src/frapi/library/Zend/Loader/PluginLoader.php on line 384

Notice: include_once(): apc_cache_find [257073455]
 in /Users/jason/frapi/src/frapi/library/Zend/Loader/PluginLoader.php on line 384

Notice: include_once(): 1. h->opened_path=[/Users/jason/frapi/src/frapi/library/Zend/Application/Resource/ResourceAbstract.php]  h->filename=[/Users/jason/frapi/src/frapi/library/Zend/Application/Resource/ResourceAbstract.php]
 in /Users/jason/frapi/src/frapi/library/Zend/Loader.php on line 146

Notice: include_once(): apc_cache_find [257073459]
 in /Users/jason/frapi/src/frapi/library/Zend/Loader.php on line 146

Notice: include_once(): 1. h->opened_path=[/Users/jason/frapi/src/frapi/library/Zend/Application/Resource/Resource.php]  h->filename=[/Users/jason/frapi/src/frapi/library/Zend/Application/Resource/Resource.php]
 in /Users/jason/frapi/src/frapi/library/Zend/Loader.php on line 146

Notice: include_once(): apc_cache_find [257073458]
 in /Users/jason/frapi/src/frapi/library/Zend/Loader.php on line 146

Notice: include_once(): 1. h->opened_path=[/Users/jason/frapi/src/frapi/library/Zend/Application/Resource/Frontcontroller.php]  h->filename=[/Users/jason/frapi/src/frapi/library/Zend/Application/Resource/Frontcontroller.php]
 in /Users/jason/frapi/src/frapi/library/Zend/Loader/PluginLoader.php on line 384

Notice: include_once(): apc_cache_find [257073450]
 in /Users/jason/frapi/src/frapi/library/Zend/Loader/PluginLoader.php on line 384

Notice: include_once(): 1. h->opened_path=[/Users/jason/frapi/src/frapi/library/Zend/Application/Module/Autoloader.php]  h->filename=[/Users/jason/frapi/src/frapi/library/Zend/Application/Module/Autoloader.php]
 in /Users/jason/frapi/src/frapi/library/Zend/Loader.php on line 146

Notice: include_once(): apc_cache_find [257073443]
 in /Users/jason/frapi/src/frapi/library/Zend/Loader.php on line 146

Notice: include_once(): 1. h->opened_path=[/Users/jason/frapi/src/frapi/library/Zend/Loader/Autoloader/Resource.php]  h->filename=[/Users/jason/frapi/src/frapi/library/Zend/Loader/Autoloader/Resource.php]
 in /Users/jason/frapi/src/frapi/library/Zend/Loader.php on line 146

Notice: include_once(): apc_cache_find [257074606]
 in /Users/jason/frapi/src/frapi/library/Zend/Loader.php on line 146

Notice: include_once(): 1. h->opened_path=[/Users/jason/frapi/src/frapi/library/Zend/Loader/Autoloader/Interface.php]  h->filename=[/Users/jason/frapi/src/frapi/library/Zend/Loader/Autoloader/Interface.php]
 in /Users/jason/frapi/src/frapi/library/Zend/Loader.php on line 146

Notice: include_once(): apc_cache_find [257074605]
 in /Users/jason/frapi/src/frapi/library/Zend/Loader.php on line 146

Notice: include_once(): 1. h->opened_path=[/Users/jason/frapi/src/frapi/library/Zend/Registry.php]  h->filename=[/Users/jason/frapi/src/frapi/library/Zend/Registry.php]
 in /Users/jason/frapi/src/frapi/library/Zend/Loader.php on line 146

Notice: include_once(): apc_cache_find [257075519]
 in /Users/jason/frapi/src/frapi/library/Zend/Loader.php on line 146

Notice: include_once(): 1. h->opened_path=[/Users/jason/frapi/src/frapi/library/Zend/Translate.php]  h->filename=[/Users/jason/frapi/src/frapi/library/Zend/Translate.php]
 in /Users/jason/frapi/src/frapi/library/Zend/Loader.php on line 146

Notice: include_once(): apc_cache_find [257076314]
 in /Users/jason/frapi/src/frapi/library/Zend/Loader.php on line 146

Notice: include_once(): 1. h->opened_path=[/Users/jason/frapi/src/frapi/library/Zend/Translate/Adapter/Gettext.php]  h->filename=[/Users/jason/frapi/src/frapi/library/Zend/Translate/Adapter/Gettext.php]
 in /Users/jason/frapi/src/frapi/library/Zend/Loader.php on line 146

Notice: include_once(): apc_cache_find [257076320]
 in /Users/jason/frapi/src/frapi/library/Zend/Loader.php on line 146

Notice: include_once(): 1. h->opened_path=[/Users/jason/frapi/src/frapi/library/Zend/Translate/Adapter.php]  h->filename=[/Users/jason/frapi/src/frapi/library/Zend/Translate/Adapter.php]
 in /Users/jason/frapi/src/frapi/library/Zend/Loader.php on line 146

Notice: include_once(): apc_cache_find [257076316]
 in /Users/jason/frapi/src/frapi/library/Zend/Loader.php on line 146

Notice: include_once(): 1. h->opened_path=[/Users/jason/frapi/src/frapi/library/Zend/Locale.php]  h->filename=[/Users/jason/frapi/src/frapi/library/Zend/Locale.php]
 in /Users/jason/frapi/src/frapi/library/Zend/Loader.php on line 146

Notice: include_once(): apc_cache_find [257074612]
 in /Users/jason/frapi/src/frapi/library/Zend/Loader.php on line 146

Notice: include_once(): 1. h->opened_path=[/Users/jason/frapi/src/frapi/library/Zend/Translate/Exception.php]  h->filename=[/Users/jason/frapi/src/frapi/library/Zend/Translate/Exception.php]
 in /Users/jason/frapi/src/frapi/library/Zend/Loader.php on line 146

Notice: include_once(): apc_cache_find [257076327]
 in /Users/jason/frapi/src/frapi/library/Zend/Loader.php on line 146

Notice: include_once(): 1. h->opened_path=[/Users/jason/frapi/src/frapi/library/Zend/Exception.php]  h->filename=[/Users/jason/frapi/src/frapi/library/Zend/Exception.php]
 in /Users/jason/frapi/src/frapi/library/Zend/Loader.php on line 146

Notice: include_once(): apc_cache_find [257073851]
 in /Users/jason/frapi/src/frapi/library/Zend/Loader.php on line 146

Notice: include_once(): 1. h->opened_path=[/Users/jason/frapi/src/frapi/library/Zend/Session/Namespace.php]  h->filename=[/Users/jason/frapi/src/frapi/library/Zend/Session/Namespace.php]
 in /Users/jason/frapi/src/frapi/library/Zend/Loader.php on line 146

Notice: include_once(): apc_cache_find [257076026]
 in /Users/jason/frapi/src/frapi/library/Zend/Loader.php on line 146

Notice: include_once(): 1. h->opened_path=[/Users/jason/frapi/src/frapi/library/Zend/Session/Abstract.php]  h->filename=[/Users/jason/frapi/src/frapi/library/Zend/Session/Abstract.php]
 in /Users/jason/frapi/src/frapi/library/Zend/Loader.php on line 146

Notice: include_once(): apc_cache_find [257076024]
 in /Users/jason/frapi/src/frapi/library/Zend/Loader.php on line 146

Notice: include_once(): 1. h->opened_path=[/Users/jason/frapi/src/frapi/library/Zend/Session.php]  h->filename=[/Users/jason/frapi/src/frapi/library/Zend/Session.php]
 in /Users/jason/frapi/src/frapi/library/Zend/Loader.php on line 146

Notice: include_once(): apc_cache_find [257076022]
 in /Users/jason/frapi/src/frapi/library/Zend/Loader.php on line 146

Notice: include_once(): 1. h->opened_path=[/Users/jason/frapi/src/frapi/library/Zend/Session/Exception.php]  h->filename=[/Users/jason/frapi/src/frapi/library/Zend/Session/Exception.php]
 in /Users/jason/frapi/src/frapi/library/Zend/Loader.php on line 146

Notice: include_once(): apc_cache_find [257076025]
 in /Users/jason/frapi/src/frapi/library/Zend/Loader.php on line 146
```

**Comment by epixa** [comment]
The fatal error is actually directly the result of the prior notices, so they are the ones at which we should be looking.  Can you add the notices to this issue?

**Comment by jasonrhodes** [comment]
Ah, excellent point. They look like they are APC related, but I installed APC (via homebrew) and added the extension=apc.so to the php.ini file, so I _thought_ I had it set up correctly. Anyway, I added the notices up in the issue text.

**Comment by epixa** [comment]
This may sound absolutely nuts, but have you restarted your computer (not just php) after installing apc?  I've have php-fpm installed via homebrew, and when I installed apc, everything seemed to go to hell when trying to access apc.  I'm sure there is a non-restart related solution, but a restart did fix the issue for me.

**Comment by jasonrhodes** [comment]
Interesting! I hadn't restarted but I just did and I still get the same errors.

**Comment by davidcoallier** [comment]
This is a bad version of APC that was accidentally rolled out. You might need to install the -beta version of the extension.

Can you tell me what's your PHP Version and more importantly your APC version?

**Comment by jasonrhodes** [comment]
Sure thing. Looks like APC 3.1.9 - and in case you need more info:

```
$ pecl info apc
About pecl.php.net/APC-3.1.9
============================
Release Type          PECL-style PHP extension (source code)
Name                  APC
Channel               pecl.php.net
Summary               Alternative PHP Cache
Description           APC is a free, open, and robust framework for
                      caching and optimizing PHP intermediate code.
Maintainers           Daniel Cowgill <dan@mail.communityconnect.com>
                      (lead, inactive)
                      George Schlossnagle <george@omniti.com> (lead,
                      inactive)
                      Rasmus Lerdorf <rasmus@php.net> (lead)
                      Gopal Vijayaraghavan <gopalv@php.net> (lead)
                      Edin Kadribasic <edink@emini.dk> (developer,
                      inactive)
                      Ilia Alshanetsky <ilia@prohost.org> (developer)
                      Marcus Börger <helly@php.net> (developer,
                      inactive)
                      Sara Golemon <pollita@php.net> (developer,
                      inactive)
                      Brian Shire <shire@php.net> (developer)
                      Kalle Sommer Nielsen <kalle@php.net> (developer)
                      Pierre Joye <pierre@php.net> (developer)
Release Date          2011-05-14 22:14:56
Release Version       3.1.9 (stable)
API Version           3.1.0 (stable)
License               PHP License (http://www.php.net/license)
Release Notes         - Fix regression introduced in 3.1.8 (#22687)
Required Dependencies PHP version 5.1.0
                      PEAR installer version 1.4.0 or newer
package.xml version   2.0
Last Modified         2011-11-11 00:22
Previous Installed    - None -
Version
```

PHP looks like version 5.3.4

```
$ php --version
PHP 5.3.4 (cli) (built: Dec 15 2010 12:15:07) 
```

**Comment by jasonrhodes** [comment]
I found a temporary way around this. I rolled APC back to version 3.1.6 :

```
$ sudo pecl install -f apc-3.1.6
```

After restarting Apache, I can access the FRAPI admin without any errors. Maybe FRAPI doesn't play well with the newer versions of APC, or maybe the newer versions of APC aren't as stable as they say? 

(FYI: I turned OFF debugging on the newer versions of APC. Some people on Server Fault and Stack Overflow had suggested that was a fix, but I never had debugging on in the first place.)
